src/rgw/rgw_rest.cc

```diff
@@ -1825,8 +1825,28 @@ static http_op op_from_method(const char *method)
 
 int RGWHandler_REST::init_permissions(RGWOp* op)
 {
-  if (op->get_type() == RGW_OP_CREATE_BUCKET)
+  if (op->get_type() == RGW_OP_CREATE_BUCKET) {
+    // We don't need user policies in case of STS token returned by AssumeRole, hence the check for user type
+    if (! s->user->user_id.empty() && s->user->type != TYPE_NONE) {
+      try {
+        map<string, bufferlist> uattrs;
```

Comment:
I don't know if it's valid, but just on the surface, I wonder if there is some encapsulation we could do to grovel less over the containers when merging the policies here?

Reply:
I'll look into this.